src/test/scala/gitbucket/core/api/ApiSpecModels.scala

```diff
@@ -42,228 +48,609 @@ object ApiSpecModels {
+       |"pull_request_url":"http://gitbucket.exmple.com/api/v3/repos/octocat/Hello-World/pulls/1347",
+       |"_links":{
+         |"self":{"href":"http://gitbucket.exmple.com/api/v3/repos/octocat/Hello-World/pulls/comments/29724692"},
+         |"html":{"href":"http://gitbucket.exmple.com/octocat/Hello-World/pull/1347#discussion_r29724692"},
+         |"pull_request":{"href":"http://gitbucket.exmple.com/api/v3/repos/octocat/Hello-World/pulls/1347"}}
+       |}""".stripMargin
+
+  val jsonComment = s"""{
+       |"id":1,
+       |"user":$jsonUser,
+       |"body":"Me too",
+       |"created_at":"2011-04-14T16:00:49Z",
+       |"updated_at":"2011-04-14T16:00:49Z",
+       |"html_url":"http://gitbucket.exmple.com/octocat/Hello-World/issues/1347#comment-1"
+       |}""".stripMargin
+
+  val jsonCommentPR = s"""{
+       |"id":1,
+       |"user":$jsonUser,
+       |"body":"Me too",
+       |"created_at":"2011-04-14T16:00:49Z",
+       |"updated_at":"2011-04-14T16:00:49Z",
+       |"html_url":"http://gitbucket.exmple.com/octocat/Hello-World/pull/1347#comment-1"
+       |}""".stripMargin
+
+  val jsonCommitListItem = s"""{
+       |"sha":"6dcb09b5b57875f334f61aebed695e2e4193db5e",
+       |"commit":{
+         |"message":"full message",
+         |"author":{"name":"octocat","email":"octocat@example.com","date":"2011-04-14T16:00:49Z"},
+         |"committer":{"name":"octocat","email":"octocat@example.com","date":"2011-04-14T16:00:49Z"},
+         |"url":"http://gitbucket.exmple.com/api/v3/repos/octocat/Hello-World/git/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e"
+       |},
+       |"parents":[{
+         |"sha":"1da452aa92d7db1bc093d266c80a69857718c406",
+         |"url":"http://gitbucket.exmple.com/api/v3/repos/octocat/Hello-World/commits/1da452aa92d7db1bc093d266c80a69857718c406"}],
+       |"url":"http://gitbucket.exmple.com/api/v3/repos/octocat/Hello-World/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e"
+       |}""".stripMargin
+
+  // TODO url is correct?
+  val jsonCommit = (id: String) => s"""{
+       |"id":"$id",
+       |"message":"full message",
+       |"timestamp":"2011-04-14T16:00:49Z",
+       |"added":[],
+       |"removed":[],
+       |"modified":["README.md"],
+       |"author":{"name":"octocat","email":"octocat@example.com","date":"2011-04-14T16:00:49Z"},
+       |"committer":{"name":"octocat","email":"octocat@example.com","date":"2011-04-14T16:00:49Z"},
+       |"url":"http://gitbucket.exmple.com/octocat/Hello-World/commit/$id"
+       |}""".stripMargin
+
+  // TODO comment_url
+  val jsonCommits = s"""{
+       |"url":"http://gitbucket.exmple.com/api/v3/repos/octocat/Hello-World/commits/6dcb09b5b57875f334f61aebed695e2e4193db5e",
+       |"sha":"6dcb09b5b57875f334f61aebed695e2e4193db5e",
+       |"html_url":"http://gitbucket.exmple.comoctocat/Hello-World/commit/6dcb09b5b57875f334f61aebed695e2e4193db5e",
+       |"comment_url":"http://gitbucket.exmple.com",
```

Comment:
Is this `comment_url` correct?

Reply:
It seems to that we don't have API for commit comment. We might have to remove this property from the model.

Reply:
ok